### PR TITLE
Database Connection and Basic User Model

### PR DESCRIPTION
**Pull request by dotslash227**
- Checks if there is a valid mongodb connection or not
- Server run fails if there is none, else runs
- Contains a basic user model/schema to be used for authentication

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,6 @@
-# EXPRESS-STARTER
+# Poll App. Hatchways Project
+
+## Server Setup
+1. Run "yarn install" or "npm install" to install all node modules needed for the app to run
+2. Make sure MongoDB is setup and running on your localhost on port 27017. The app will connec to "poll" database on your local mongo connection.
```
Comment (tgnemecek) [inline README.md]:
```suggestion
2. Make sure MongoDB is setup and running on your localhost on port 27017. The app will connect to "poll" database on your local mongo connection.
```

**Review by tgnemecek — APPROVED** [review]

File: server/app.js [hunk 2]
```diff
@@ -10,13 +10,24 @@ const pingRouter = require("./routes/ping");
 const { json, urlencoded } = express;
 
 var app = express();
+var mongoose = require("mongoose"); //For database connection
+
+// Connecting to database. fail if not able to connect
+mongoose.connect("mongodb://localhost:27017", {useNewUrlParser:true});
+var db = mongoose.connection;
```
Comment (skassam21) [inline server/app.js]: I know we used var above in our code, but let's consistently use either let or const. We want to favor using const over let (especially if we never need to reuse it) - so in these cases we want to use const

File: server/yarn.lock [hunk 3]
```diff
@@ -169,6 +169,22 @@ bindings@^1.5.0:
...
+    readable-stream "^2.3.5"
+    safe-buffer "^5.1.1"
+
+bluebird@3.5.1:
+  version "3.5.1"
+  resolved "https://registry.yarnpkg.com/bluebird/-/bluebird-3.5.1.tgz#d9551f9de98f1fcda1e683d17ee91a0602ee2eb9"
+  integrity sha512-MKiLiV+I1AA596t9w1sQJ8jkiSr5+ZKi0WKrYGUn6d1Fx+Ij4tIj+m2WMQSGczs5jZVxV339chE8iwk6F64wjA==
+
+=======
+>>>>>>> 52071bae80408d5c987f58ea1f27142a1de3962c
```
Comment (skassam21) [inline server/yarn.lock:187]: Address these merge conflicts - even in the yarn.lock file

**Review by skassam21 — COMMENTED** [review]
Looks good

File: server/app.js (hunk 2 above)
Comment (dotslash227) [reply server/app.js]: Understood. Updated var to const.

File: server/yarn.lock (hunk 3 above)
Comment (dotslash227) [reply server/yarn.lock:187]: Had addressed the conflicts in morning.

**Review by Hakeemmidan — COMMENTED** [review]
Nice. One quick note that I noticed is:
- My terminal gave me this warning when I ran the server: 
`
DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
`
  - I changed line 16 `server/app.js` to `mongoose.connect("mongodb://localhost:27017", { useUnifiedTopology: true, useNewUrlParser: true });` and it fixed it

**Pull request merged**